### PR TITLE
Bash cleanup

### PR DESCRIPTION
**Pull request by Mrnoname1000**
Here are a bunch of changes cleaning up the Bash launcher. Most of it is assisted by [shellcheck](https://www.shellcheck.net).
Changes include syntax improvements, removing unused variables, and some minimal syntax changes to make it easier to port to POSIX shell in the future.
See commit messages for more details.

### CODE REVIEW

**Comment by enebo** [comment]
If there ends up being some esoteric platform which somehow disagrees with any of this we should land it sooner than later.

**Comment by headius** [comment]
I discussed with @mrnoname1000 the remaining steps for POSIX shell support and most of them boil down to leveraging sh argument arrays to replace the += bash arrays we use heavily. Historically the main trouble we had with sh was losing proper quoting when forced to reinterpolate arguments during arg processing. The other challenge is that there's no readlink in posix so we'd either have to require that (probably not a big deal) or import a non-trivial sh-based implementation of it. Thanks for the deep investigation @mrnoname1000!

**Comment by enebo** [comment]
@headius @Mrnoname1000 sounds great.  Using simplest sh that is portable sounds like a dream to me :)

File: bin/jruby.bash [hunk 1]
```diff
@@ -125,41 +115,40 @@ add_log "  JAVA_OPTS: $JAVA_OPTS"
...
-    JAVA_HOME=$(dirname $(dirname $expanded_javacmd))
+  expanded_javacmd="$(command -v "$JAVACMD")"
+  if [ -z "$JAVA_HOME" ] && [ -x "$expanded_javacmd" ] ; then
+    JAVA_HOME="$(dirname "$(dirname "$expanded_javacmd")")"
   fi
 fi
 
 # Detect Java 9+ by the presence of a jmods directory in JAVA_HOME
-if [ -d $JAVA_HOME/jmods ]; then
+if [ -d "$JAVA_HOME"/jmods ]; then
```
Comment (eregon) [inline bin/jruby.bash]: Note: this actually fixes a warning running on Windows in a Bash shell:
```
/c/Users/runneradmin/.rubies/jruby-9.2.9.0/bin/jruby: line 155: [: C:\Program: binary operator expected
```
This happens in GitHub Actions.

Comment (headius) [reply bin/jruby.bash]: Makes sense with spaces in path.

Comment (eregon) [reply bin/jruby.bash]: Yeah, exactly. Using the Bash launcher Windows is likely unusual though (Just `jruby` would give precedence to the `.exe`, only `/full/path/to/jruby` or `jruby.bash` in a Bash shell would use the Bash launcher).

**Pull request merged**